creator-node/src/serviceRegistry.js

```diff
@@ -58,6 +58,25 @@ class ServiceRegistry {
     this.monitoringQueue.start()
   }
 
+  /**
+  * Initializes the blacklistManager if it is not already initialized, and then returns it
+  * @returns initialized blacklistManager instance
+  */
+  async getBlacklistManager () {
+    if (!this.blacklistManager.initialized) {
```

Comment:
is it dangerous for blacklist manager to be called to .init() multiple times? probably fine, but just calling out that if multiple services try to call `getBlacklistManager` at the same time, they'll both enter the init block

Reply:
if BM gets initialized, then it grabs the contentbl db entries and adds to redis. seems potentially dangerous? should we implement some lock mechanism?

Reply:
@dmanjunath

Reply:
you could have a this.initializing field and in .init() just check to see if !initializing

Reply:
adding initializing field

Reply:
for this init op, shouldnt need a lock. since all it does is pull from db and add to a set in redis, should be fine

Reply:
hmmmm ok i'll omit adding an init lock then